src/rank_llm/analysis/response_analysis_verbose.py

```diff
@@ -2,37 +2,63 @@
 import json
 import os
 import re
-from typing import List, Dict
-
 import sys
-import os
+from typing import Dict, List, Tuple, Union
 
 SCRIPT_DIR = os.path.dirname(os.path.abspath(__file__))
 parent = os.path.dirname(SCRIPT_DIR)
 parent = os.path.dirname(parent)
 sys.path.append(parent)
 
-from rank_llm.rerank.rankllm import PromptMode
+from rank_llm.result import Result
+
+
+# For loading in list of Results
+def load_json_input_analyzer(input_json: str) -> Union[List[str], List[Result]]:
```

Comment:
I don't understand why this is needed and why the input_json is of type str?
The current input format for response analyzer is a list of file names, it opens them and reads from them. What I want to add is for response analyzer to be able to get a list[results] as input and analyze them. Ultimately what I want in the demo files is to have:
```
retriever = Retriever.from...
retrieved_ results = retriever.retrieve(...)
reranker = ZephyrReranker()
rerank_results = reranker.rerank(..)
response_analyzer = ResponseAnalyzer.from_inline_results(rerank_results)
response_error = response_anaylzer.count_errors()